endpoints/s3/src/getobject.cpp

```diff
@@ -113,8 +112,33 @@ void irods::s3::actions::handle_getobject(
         return;
     }
 
-    std::shared_ptr<persisent_data> persistent_data_ptr =
-        std::make_shared<persisent_data>(*irods_username, path);
+    const auto& config = irods::http::globals::configuration();
+    const auto& irods_client_config = config.at("irods_client");
+    const auto& zone = irods_client_config.at("zone").get_ref<const std::string&>();
+
+    const auto& rodsadmin_username =
+        irods_client_config.at(json_pointer{"/proxy_admin_account/username"}).get_ref<const std::string&>();
+    auto rodsadmin_password =
+        irods_client_config.at(json_pointer{"/proxy_admin_account/password"}).get_ref<const std::string&>();
+
+    auto conn = std::make_shared<irods::experimental::client_connection>(
+        irods::experimental::defer_authentication,
+        irods_client_config.at("host").get_ref<const std::string&>(),
+        irods_client_config.at("port").get<int>(),
+        irods::experimental::fully_qualified_username{rodsadmin_username, zone},
+        irods::experimental::fully_qualified_username{*irods_username, zone});
+
+    auto* conn_ptr = static_cast<RcComm*>(*conn);
+
+    if (const auto ec = clientLoginWithPassword(conn_ptr, rodsadmin_password.data()); ec < 0) {
+        logging::error("{}: clientLoginWithPassword error: {}", __func__, ec);
+        response.result(beast::http::status::internal_server_error);
+        session_ptr->send(std::move(response));
+        return;
+    }
```

Comment:
Let's write up an issue to move this into a function so it can be available across the application.

We can handle it after 0.2.0.

Reply:
There is one called get_connection() in common.cpp.  It is very similar.  The problem is that it does not return a shared_ptr so if it is returned it will go out of scope before intended.

If I remember correctly, that conn is not moveable.

Reply:
The connection_facade is designed to hold either a `client_connection` or a `connection_pool::connection_proxy`. It makes dealing with 4.2 and 4.3 servers easier (i.e. pooling vs no pooling).

As for the code I referenced, you can create a function called `make/create_dedicated_irods_connection(string_view _username)`. It would be for cases where the amount of time the connection is needed could be very long. That keeps the connection pool's connections free for short running requests.